### PR TITLE
Refactor Capsule -> Atom & Add Kubernetes Support

### PR DESCRIPTION
**Pull request by rocketbitz**
`ok      github.com/caesium-dev/caesium/internal/atom/kubernetes (cached)        coverage: 92.7% of statements`

### CODE REVIEW

File: internal/atom/kubernetes/atom.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	if c.metadata.Status.StartTime == nil {
+		return time.Time{}
+	}
+	return c.metadata.Status.StartTime.Time
+}
+
+// StoppedAt returns the UTC time the Atom was stopped.
+func (c *Atom) StoppedAt() time.Time {
+	return c.metadata.DeletionTimestamp.Time
+}
```
Comment (RohanDalton) [inline internal/atom/kubernetes/atom.go:60]: I'm not sure that I understand the logic here. Under what circumstances would the started time be `nil`, but we would want the returned value to be the current time? Specifically in the case that the atom had been created but not started wouldn't we want/expect the returned value to be `nil`?

File: internal/atom/kubernetes/engine.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+// for interacting with kubernetes.Atoms.
+func NewEngine(ctx context.Context) Engine {
+	k8sCfg := env.Variables().KubernetesConfig
+	if k8sCfg == "" {
+		u, _ := user.Current()
+		k8sCfg = filepath.Join(u.HomeDir, kubeConfig)
+	} else {
+		k8sCfg = filepath.Join(k8sCfg, kubeConfig)
+	}
+
```
Comment (RohanDalton) [inline internal/atom/kubernetes/engine.go:40]: Where is `kubeConfig` coming from?

**Review by RohanDalton — APPROVED** [review]

File: internal/atom/kubernetes/engine.go (hunk 2 above)
Comment (rocketbitz) [reply internal/atom/kubernetes/engine.go:40]: It's a constant in another file. I need to still refactor this a bit so the unit tests work if there is no ~/.kube/config file. They ran on my machine but not in CircleCI.

File: internal/atom/kubernetes/atom.go (hunk 1 above)
Comment (rocketbitz) [reply internal/atom/kubernetes/atom.go:60]: time.Now() is current time, time.Time{} is the zero time. We're not returning a pointer here so can't return nil.

File: internal/atom/kubernetes/kubernetes.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+		1:   atom.Failure,
+		125: atom.StartupFailure,
+		126: atom.StartupFailure,
+		127: atom.StartupFailure,
+		137: atom.Killed,
+		143: atom.Terminated,
+	}
+)
+
+const kubeConfig = ".kube/config"
```
Comment (rocketbitz) [inline internal/atom/kubernetes/kubernetes.go:32]: It's here @RohanDalton

**Pull request merged**